Sources/Identity/CustomerInfoManager.swift

```diff
@@ -321,35 +321,33 @@ private extension CustomerInfoManager {
             _ = Task<Void, Never> {
                 let transactions = await self.transactionFetcher.unfinishedVerifiedTransactions
 
-                if !transactions.isEmpty {
-                    var results: [Result<CustomerInfo, BackendError>] = []
-                    let storefront = await Storefront.currentStorefront
-
+                if let transactionToPost = transactions.first {
                     Logger.debug(
                         Strings.customerInfo.posting_transactions_in_lieu_of_fetching_customerinfo(transactions)
                     )
 
-                    for transaction in transactions {
-                        results.append(
-                            await self.transactionPoster.handlePurchasedTransaction(
-                                transaction,
-                                data: .init(appUserID: appUserID,
-                                            presentedOfferingID: nil,
-                                            unsyncedAttributes: [:],
-                                            storefront: storefront,
-                                            source: Self.sourceForUnfinishedTransaction)
-                            )
-                        )
-                    }
+                    let transactionData = PurchasedTransactionData(
+                        appUserID: appUserID,
+                        presentedOfferingID: nil,
+                        unsyncedAttributes: [:],
+                        storefront: await Storefront.currentStorefront,
+                        source: Self.sourceForUnfinishedTransaction
+                    )
 
-                    // Any of the POST receipt operations will have posted the same receipt contents
-                    // so the resulting `CustomerInfo` will be equivalent.
-                    // For that reason, we can return the last known success if available,
-                    // and otherwise the last result (an error).
-                    let lastSuccess = results.last { $0.value != nil }
-                    let result = lastSuccess ?? results.last!
+                    // Post everything but the first transaction in the background
+                    // in parallel so they can be de-duped
```

Comment:
will the first and the second one be de-duped with this approach?

Reply:
Yeah (we start the last N-1 in the background and then immediately start the first one), but let me make sure that's covered in a test.